components/jaggery-core/org.jaggeryjs.jaggery.core/src/main/java/org/jaggeryjs/jaggery/core/manager/JaggeryDeployerManager.java

```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jaggeryjs.jaggery.core.manager;
+
+import org.apache.catalina.*;
+import org.apache.catalina.startup.Tomcat;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.tomcat.util.descriptor.web.*;
+import org.jaggeryjs.hostobjects.log.LogHostObject;
+import org.jaggeryjs.jaggery.core.JaggeryCoreConstants;
+import org.jaggeryjs.jaggery.core.ScriptReader;
+import org.jaggeryjs.scriptengine.cache.ScriptCachingContext;
+import org.jaggeryjs.scriptengine.engine.JaggeryContext;
+import org.jaggeryjs.scriptengine.engine.RhinoEngine;
+import org.jaggeryjs.scriptengine.exceptions.ScriptException;
+import org.jaggeryjs.scriptengine.util.HostObjectUtil;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.mozilla.javascript.ScriptableObject;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * This JaggeryDeployerManager is responsible for processing jaggery.conf file and making relevant modifications
+ * according to jaggery.conf file
+ */
+public class JaggeryDeployerManager {
+
+    private static Log log = LogFactory.getLog(JaggeryDeployerManager.class);
+    private static final String INDEX_HTML = "index.html";
+    private static final String INDEX_JAG = "index.jag";
+
+    /**
+     * Process the jaggery.conf file of jaggery apps
+     * @param context context of the jaggery app
+     */
+    public void processJaggeryApp(Context context) {
```

Comment:
make this also as a static method as others, as this particular class doesn't have a state, and its just a engine to process the application.